test/http/HttpRequest.test.ts

```diff
@@ -11,6 +11,34 @@ import { HttpRequest } from '../../src/http/HttpRequest';
 chai.use(chaiAsPromised);
 
 describe('HttpRequest', () => {
+    it('clone', async () => {
+        const req = new HttpRequest({
+            method: 'POST',
+            url: 'http://localhost:7071/api/helloWorld',
+            body: {
+                string: 'body1',
+            },
+            headers: {
+                a: 'b',
+            },
+            params: {
+                c: 'd',
+            },
+            query: {
+                e: 'f',
+            },
+        });
+        const req2 = req.clone();
+        expect(await req.text()).to.equal('body1');
+        expect(await req2.text()).to.equal('body1');
+        expect(req.headers).to.not.equal(req2.headers);
+        expect(req.headers).to.deep.equal(req2.headers);
+        expect(req.params).to.not.equal(req2.params);
+        expect(req.params).to.deep.equal(req2.params);
+        expect(req.query).to.not.equal(req2.query);
```

Comment:
Nit: can we have a newline between the different sections? It would make it a little easier to follow what's being tested in each pair.